File: test_digest.js
const md5 = require('md5');

const ha1 = md5('admin:Authentication Required:password');
const ha2 = md5('POST:/upload');
const response = md5(`${ha1}:f6e94ba6ed83fbcc3613eecffc70dd7d:00000001:7d53a137caf430c6:auth:${ha2}`);

console.log('HA1:', ha1);
console.log('HA2:', ha2);
console.log('Response:', response);

File: mock_server/test_digest.py
import hashlib

ha1 = hashlib.md5(b'admin:Authentication Required:password').hexdigest()
ha2 = hashlib.md5(b'POST:/upload').hexdigest()
nonce = 'f6e94ba6ed83fbcc3613eecffc70dd7d'
nc = '00000001'
cnonce = '7d53a137caf430c6'
qop = 'auth'
response = hashlib.md5(f'{ha1}:{nonce}:{nc}:{cnonce}:{qop}:{ha2}'.encode('utf-8')).hexdigest()
print('Expected Response Python:', response)
